### PR TITLE
BookRepository start extends JpaRepository. Update and delete methods…

### PR DESCRIPTION
**Pull request by slizko1**
… and Liquibase was added. spring.jpa.hibernate.ddl-auto property value was changed to validate.  soft delete concept for the Book entity was implemented.

### CODE REVIEW

**Comment by AnyaRatusnyak** [comment]
Well done ))

File: src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -30,7 +34,18 @@ public BookDto getBookById(@PathVariable Long id) {
...
+    public BookDto createBook(@RequestBody BookRequestDto bookDto) {
         return bookService.createBook(bookDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto update(@PathVariable Long id, @RequestBody BookRequestDto bookRequestDto) {
+        return bookService.update(id, bookRequestDto);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (karma-o) [inline src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java:47]: Let's also add `created` response status for `createBook()` method

File: src/main/java/com/samoilenko/onlinebookstore/model/Book.java [hunk 2]
```diff
@@ -10,11 +10,15 @@
...
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
 @EqualsAndHashCode
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (karma-o) [inline src/main/java/com/samoilenko/onlinebookstore/model/Book.java]: where is deprecated so let's use `@SqlRestriction` instead, if it doesn't work - try to change the type of `is_deleted` column in liquibase to `BIT`

File: src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -37,4 +37,16 @@ public BookDto findBookById(Long id) {
...
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, BookRequestDto bookRequestDto) {
+        Book updatedBook = bookMapper.toModel(bookRequestDto);
+        updatedBook.setId(id);
+        return bookMapper.toDto(bookRepository.save(updatedBook));
+    }
```
Comment (karma-o) [inline src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java]: first, we need to check if the book with such an id exists, if not - throw `EntityNotFountException`, and than act, fix for both methods

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+                    unique: true
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
```
Comment (karma-o) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: description and URL's of the image are usually pretty long, let's raise the max length to at least 500

**Review by karma-o — CHANGES_REQUESTED** [review]
Looks good so far, left several suggestions

File: src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -39,11 +39,16 @@ public BookDto findBookById(Long id) {
 
     @Override
     public void deleteById(Long id) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Book with id " + id + " not found");
+        }
```
Comment (karma-o) [inline src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java:44]: code duplication, extract into a private method

**Review by karma-o — APPROVED** [review]
GJ!

**Pull request merged**